flink-kubernetes/src/main/java/org/apache/flink/kubernetes/utils/KubernetesUtils.java

```diff
@@ -83,6 +85,8 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(KubernetesUtils.class);
 
+    private static final YAMLMapper yamlMapper = new YAMLMapper();
```

Comment:
Where does this dependency come from? I think it is pulled in from `io.fabric8:kubernetes-client`. If we need a dependency then we should add it as a dependency to `pom.xml` and not rely on transitive dependencies which might change when changing the direct dependency.

Reply:
Maybe it is also better to rely on the shaded jackson dependency.

Reply:
You are right. The `jackson` dependency is pulled in via `io.fabric8:kubernetes-client`. And we have shaded these classes in `maven-shade-plugin`. 

I also hesitated to directly rely on the transitive dependencies. Compared with explicitly adding the dependency, I prefer to rely on the shaded jackson.

Reply:
I think we should still add the `flink-shaded-jackson` dependency explicitly to the `pom.xml` of the `flink-kubernetes` module because otherwise it is not self-contained. What we can do is to add

```
<dependency>
	<groupId>org.apache.flink</groupId>
	<artifactId>flink-shaded-jackson</artifactId>
	<scope>provided</scope>
</dependency>
```

which says that some other component needs to provide this dependency. But it states that this component needs this dependency in order to work.

Reply:
Thanks. I learned a lesson here about the transitive and provided dependencies.